spec/dates_spec.rb

```diff
@@ -0,0 +1,22 @@
+require_relative 'spec_helper'
+
+describe "Dates class" do
+  before do
+    @date = '9-10-2018'
+    @date_range = {begin_date: '9-10-2018', end_date: '10-10-2018'}
```

Comment:
You're missing tests for many of the methods you wrote in `Dates`. For example, your code to compare two date ranges should have a whole battery of tests here (see my note in Slack).

One of the big advantages of moving this logic into a separate class is that it allows you to test it in isolation without all the context of rooms and reservations.